include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -123,8 +121,7 @@ struct __parallel_transform_reduce_sub_group_submitter<__sub_group_size, __iters
         sycl::event __reduce_event = __exec.queue().submit([&, __n](sycl::handler& __cgh) {
             oneapi::dpl::__ranges::__require_access(__cgh, __rngs...); // get an access to data under SYCL buffer
             sycl::accessor __res_acc{__res, __cgh, sycl::write_only, __dpl_sycl::__no_init{}};
-            ::std::size_t __local_mem_size = std::max(__transform_pattern.local_mem_req(__sub_group_size),
-                                                      __reduce_pattern.local_mem_req(__sub_group_size));
+            ::std::size_t __local_mem_size = __reduce_pattern.local_mem_req(__sub_group_size);
```

Comment:
It's not a huge deal, but this feels a little dirty. In the `reduce_over_sub_group` case, this should return zero no matter what. In the `reduce_over_group` case, it should return the work-group size.

Again, it has no ill-effect here, but someone reading the code might infer some *meaning* from the call to `local_mem_req(__sub_group_size)` when really there is none.

I'd be happy with a `TODO` for now.

Reply:
I realise this is fundamentally my fault for being 'clever' and introducing the `local_mem_req` method in the first place. I think the TODO could just be to remove that method & calculate it 'by hand'...

Reply:
I have just set it to 0 and removed `local_mem_req` for sub_group reduce